components/inputs/input-number.js

```diff
@@ -264,14 +264,13 @@ class InputNumber extends LabelledMixin(SkeletonMixin(FormElementMixin(LocalizeC
 			countDecimalDigits(this._valueTrailingZeroes, false),
 			this.maxFractionDigits
 		);
-		let valueTrailingZeroes = this.value.toString();
```

Comment:
The issue stems from `this.value` being `Number` and its string representation being `5e-7` at this point.